### PR TITLE
Send flow UI update

### LINKED ISSUE #8862: Remove Recents under 'Transfer  between my accounts'
**Describe the bug**
I mistakenly sent my tokens to the address listed under Recents. Some address could not belong to you. Please remove the Recents feature

**To Reproduce (REQUIRED)**
Just send a token and pick from the Recents (some address that you sent to that does not belong to you)

**Expected behavior**

metamask

### LINKED ISSUE #18972: Send flow UI updates
## Background
Users are getting confused sending funds to what they think is "Transfer between my accounts" but sending them to recent addresses. The current UI layout is confusing and could lead to loss of funds. Reports [here](https://twitter.com/3pa15/status/1648794035094450188?s=20), and related to this [issue](https://github.com/MetaMask/MetaMask-planning/issues/398).

Current send flow UI

## Functional requirements
The contacts that appear during the send flow will differ depending on whether the user has saved contacts vs no saved contacts.

### Saved contact exists
In this scenario, what appears below the send flow will differ depending on the user's number of accounts. If the user has more than one account:
- Display list of "Your accounts"
- Display "Contacts" in alphabetical order after the final user's account

If the user has only one account, then list contacts under "Contacts" without any account listing.

If the user doesn't have saved contacts, the area below the "To" send address will be empty.

If the user has more than one account, but no contacts saved, then only accounts will be listed under "Your accounts"

## Designs
[Figma](https://www.figma.com/file/bsJPY9bMMozmdk3uk8mlxh/Send-Remove-recent-addresses?node-id=1-31372&t=PbMkzAkNRbFM5i8N-0)

### PR DESCRIPTION
**Pull request by NidhiKJha**
This PR is to update the send flow UI. The following updates have been made to the Send Flow UI:

- Remove Recents
- Display a list of "Your accounts" if the user has more than one account
- Display "Contacts" in alphabetical order after the final user's account
- UI Updates in Contacts Page (AddressBook) in Settings Page

* Fixes #18972 
* Fixes #8862 

### Before
Screenshot 2023-06-06 at 5 08 22 PM

### After

https://github.com/MetaMask/metamask-extension/assets/39872794/e4afb3e4-0f0e-4060-b8cf-d8b68036208a

## Manual Testing Steps

- Go to home screen and click on send button

- In the send list 
1.     Check there are no recents
2.     List of accounts is there if user has more than one account
3.     List of contacts

- In the Contacts Page
1.     New UI for Contacts
2.     Add Account Button is in the bottom

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

File: ui/pages/settings/contact-list-tab/contact-list-tab.component.js [hunk 1]
```diff
@@ -105,35 +104,25 @@ export default class ContactListTab extends Component {
...
-          onClick={() => {
-            history.push(CONTACT_ADD_ROUTE);
-          }}
-        >
-          {this.context.t('addContact')}
-        </Button>
-      </div>
+      <ButtonPrimary
+        className={classnames({
+          'address-book-add-button__button': true,
```
Comment (darkwing) [inline ui/pages/settings/contact-list-tab/contact-list-tab.component.js:109]: If a classname should always be added (in this case `true`), you should do:

```
className={classnames('address-book-add-button__button', { ....
```

**Review by darkwing — CHANGES_REQUESTED** [review]

File: ui/pages/settings/contact-list-tab/contact-list-tab.component.js (hunk 1 above)
Comment (NidhiKJha) [reply ui/pages/settings/contact-list-tab/contact-list-tab.component.js]: Updated here https://github.com/MetaMask/metamask-extension/pull/19465/commits/0348f3a5a0a5b4c325fcc52a02a11b3d2c645789#diff-cdd29e9c6dd899bd11506911a21f0ad3d55b5a11dd0389357d05440168547e90R107-R108

**Review by darkwing — CHANGES_REQUESTED** [review]
Functionally this is good but can we do something with the hover effect?  It looks bad, needs padding or something else.  @SaraCheikh Thoughts?
SCR-20230607-nykv

**Comment by SaraCheikh** [comment]
Hi @darkwing and @NidhiKJha the cell should be full width with 16px inner padding, so the hover effect should look like it;s taking the whole cell: 
Screenshot 2023-06-08 at 13 04 53

**Comment by NidhiKJha** [comment]
> Hi @darkwing and @NidhiKJha the cell should be full width with 16px inner padding, so the hover effect should look like it;s taking the whole cell:

Updated here https://github.com/MetaMask/metamask-extension/pull/19465/commits/49dee4f2555603022c84f870f51ee242352251b5

**Review by vthomas13 — APPROVED** [review]

**Comment by darkwing** [comment]
Much better!

**Review by darkwing — APPROVED** [review]

**Pull request merged**